### PR TITLE
cleanup(tests): Move VM subresource tests without run strategy and drop redundant tests

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Move the VM subresource tests without run strategy from tests/vm_test.go
into tests/subresource_api_test.go, clean them up a little and use the
KubeVirt client instead of virtctl through a NewRepeatableVirtctlCommand
where applicable.

Drop "[test_id:6310]should start a VirtualMachineInstance in paused
state" which is already covered by "[test_id:4597]should signal paused
state with condition" in tests/pausing_test.go.

The goal of this and further PRs is to reduce the amount of unnecessary uses of
virtctl in the e2e tests to stabilize tests, to reduce the number of flakes and to
remove redudant tests.

Before this PR:

NewRepeatableVirtctlCommand is used unnecessarily.

After this PR:

The KubeVirt client is used instead.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
/hold

Need to investigate flakes

**Comment by alicefr** [comment]
All great!
/approve

File: tests/subresource_api_test.go [hunk 1]
```diff
@@ -259,40 +258,156 @@ var _ = Describe("[sig-compute]Subresource Api", decorators.SigCompute, func() {
...
+
+			By("Ensuring the VirtualMachineInstance is removed")
+			Eventually(ThisVMIWith(vm.Namespace, vm.Name), 240*time.Second, 1*time.Second).ShouldNot(Exist())
+
+			By("Ensuring a second invocation should fail")
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, &v1.StopOptions{})
+			Expect(err).To(MatchError(ContainSubstring("VM is not running")))
+		})
+
+		It("[test_id:3007][QUARANTINE] Should force restart a VM with terminationGracePeriodSeconds>0", decorators.Quarantine, func() {
```
Comment (jcanocan) [inline tests/subresource_api_test.go:333]: Any fair chance of getting this out of quarantine?

**Review by jcanocan — COMMENTED** [review]
Thanks for your efforts! Nice job!
Just a silly comment. I understand that it is not an easy task and out of the scope of this PR. So feel free to ignore it.

File: tests/subresource_api_test.go (hunk 1 above)
Comment (0xFelix) [reply tests/subresource_api_test.go:333]: I tried getting that out of quarantine but it wasn't as easy unfortunately. Needs to be a follow up.

Comment (jcanocan) [reply tests/subresource_api_test.go:333]: I understand. Thanks for the additional context.

**Pull request merged**